### PR TITLE
Span Links Beta Documentation

### PR DESCRIPTION
**Pull request by brett0000FF**
**Updates**
* Added [new tab](https://docs-staging.datadoghq.com/brett0000FF/span-links/tracing/trace_explorer/trace_view/?tab=spanlinksbeta) to trace_view docs.
* Added [Adding span links section](https://docs-staging.datadoghq.com/brett0000FF/span-links/tracing/trace_collection/custom_instrumentation/php/?tab=currentspan#adding-span-links-beta) to PHP custom instrumentation docs.
* Added [Adding span links section](https://docs-staging.datadoghq.com/brett0000FF/span-links/tracing/trace_collection/otel_instrumentation/java/#requirements-and-limitations) to ~Java custom~ Java OpenTelemetry instrumentation docs.
* Added [Span Links page](https://docs-staging.datadoghq.com/brett0000FF/span-links/tracing/trace_collection/span_links/) under Sending Traces to Datadog.
- [x]  TODO: Need better screenshot for trace view.
- [x]  TODO: Need better code example for PHP.
- [x]  TODO: Need better code example for Java. Edit: linking to OTel docs for Java.
- [x]  TODO: Create new page with Span Link use cases under Sending Traces to Datadog.
- [x]  TODO: Link from Trace View to how-to.
- [x]  TODO: Clarify sentence about "Linked spans were not ingested" For example, linked spans are displayed only when the corresponding spans has been ingested and indexed (by retention filter for instance).

### What does this PR do? What is the motivation?

Support for span links is now available (beta).

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/tracing/trace_collection/custom_instrumentation/java.md [hunk 1]
```diff
@@ -250,6 +250,25 @@ class SomeClass {
     }
 }
 ```
+### Adding span links (Beta)
+
+<div class="alert alert-info">Support for span links is in beta and requires the <a href="https://github.com/DataDog/dd-trace-java/releases/tag/v1.24.1">Java tracer v.1.24.1+</a></div>
```
Comment (drichards-87) [inline content/en/tracing/trace_collection/custom_instrumentation/java.md:255]:
```suggestion
<div class="alert alert-info">Support for span links is in beta and requires the <a href="https://github.com/DataDog/dd-trace-java/releases/tag/v1.24.1">Java tracer v.1.24.1+</a>.</div>
```

**Review by drichards-87 — APPROVED** [review]
Looks good! Just had one very minor suggestion.

File: content/en/tracing/trace_collection/custom_instrumentation/php.md [hunk 2]
```diff
@@ -417,6 +417,44 @@ function doRiskyThing() {
...
+$spanA = \DDTrace\start_trace_span();
+$spanA->name = 'spanA';
+$distributedTracingHeaders = \DDTrace\generate_distributed_tracing_headers();
+\DDTrace\close_span();
+
+$spanB = \DDTrace\start_trace_span();
+$spanB->name = 'spanB';
+$spanB->links[] = $spanA->getLink();
+\DDTrace\close_span();
+```
```
Comment (PROFeNoM) [inline content/en/tracing/trace_collection/custom_instrumentation/php.md:454]: I think this is the code snippet for "To add a span link from an existing span:", and vice-versa :smiley:

**Review by PerfectSlayer — COMMENTED** [review]
A quick call out that `[OpenTelemetry Context propagation][1] | [Datadog distributed header format][9] is used instead.` is no more true.
It will be supported by default from `1.24.0`. Otherwise, it needs to be setup with the provided link.

**Pull request merged**